test/functional/legacy/061_undo_tree_spec.lua

```diff
@@ -0,0 +1,260 @@
+-- Tests for undo tree and :earlier and :later.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source, eq, eval, clear, execute, expect, wait =
+  helpers.feed, helpers.insert, helpers.source, helpers.eq, helpers.eval,
+  helpers.clear, helpers.execute, helpers.expect, helpers.wait
+
+local function expect_empty_buffer()
+  -- The space will be removed by helpers.dedent but is needed as dedent will
+  -- throw an error if it can not find the common indent of the given lines.
+  return expect(' ')
+end
+local function expect_line(line)
+  return eq(line, eval('getline(".")'))
+end
+local function write_file(name, text)
+  local file = io.open(name, 'w')
+  file:write(text)
+  file:flush()
+  file:close()
+end
+
+describe('undo:', function()
+  before_each(clear)
+  teardown(function()
+    os.remove('Xtest.source')
```

Comment:
Since this is removed in a teardown(), would be a bit clearer to create it setup().